Restructure API Code

ajwgeek opened the pull request:
### Working Issue
Fixes #51 and #60 

### Test
API calls and results are shown in [Postman](https://documenter.getpostman.com/view/5500678/RWgm2fwS#a941b53e-5449-4789-9ad0-ede24efd60e5). All endpoints are performing as expected.

### Major Changes
- Decouple Board, Post, and Comment models from MongoDB
- Add complete documentation to the Backend
- Update Django
- Add SimpleJWT authentication
- Add migrations to Proc file
- Fix PEP 8 warnings

@I-Smith @Ryaguy15

Commit:
Remove experimental web client

Commit:
Fix whitespace issue in ApiConfig

Commit:
Add api/migrations to .gitignore

Commit:
Update Django to 1.11.8

Commit:
Decouple Board, Post, Comment models from MongoDB

Commit:
Add SimpleJWT authentication

Commit:
Add ownership permissions to destructive endpoints

Commit:
Fix Table of Contents links in README.md

Commit:
Create CRUD viewset class

Commit:
Fix PEP 8 warnings

Commit:
Create models package

Commit:
Cleanup admin.py

Commit:
Create permissions package

Commit:
Create serializers package

Commit:
Cleanup imports in PostViewSet and CommentViewSet

Commit:
Create views package

Commit:
Create views package

Commit:
Cleanup imports in BoardSerializer, CommentSerializer, PostSerializer

Commit:
Cleanup imports in BoardViewSet, CommentViewSet, PostViewSet

Commit:
Removed unnecessary arguments from BoardViewSet and PostViewSet

Commit:
Added documentation to Board

Commit:
Fixed spelling error in Board

Commit:
Added documentation to Post

Commit:
Added docuementation to Owner attribute in Post

Commit:
Added documentation to Comment

Commit:
Update Comment and Post documentation

Commit:
Update Comment and Post documentation

Commit:
Add documentation to admin.py

Commit:
Add documentation to CreateRetrieveUpdateDestroy

Commit:
Add documentation to serializers

Commit:
Add documentation to IsOwnerOrReadOnly

Commit:
Organize imports

Commit:
Document BoardViewSet, CommentViewSet, PostViewSet

Commit:
Update CommentViewSet and PostViewSet documentation

Commit:
Fix crash when listing Posts and Comments

Commit:
Cleanup imports in urls.py

Commit:
Update .gitignore

Commit:
Update CRUD ViewSet documentation

Commit:
Add sanity check to IsOwnerOrReadOnly

Commit:
Cleanup author ForeignKey in Post and Comment

Commit:
Cleanup imports on PostViewSet

Commit:
Add migrations to Proc file

Commit:
Create UserSerializer

Commit:
Create UserViewSet

Commit:
Update IsOwnerOrReadOnly to support Django Users

Commit:
Fix PEP 8 warnings

Ryaguy15 commented:
Couple of things. Are we getting rid of the alias idea? Should the django secret key be in the repo?

Commit:
Check in alias.py

Commit:
Fix PEP 8 warnings in alias.py

Commit:
Added environment variable for SECRET_KEY and instructions

ajwgeek commented:
Great observations, @Ryaguy15. 

I forgot to check the Alias Generator into Git. I fixed some of the PEP 8 warnings within it, but not all of them. I will be opening a separate issue about that. 

I've also added an environment variable for Django's secret key. Instructions for Heroku have been added to the README.

Ryaguy15 commented:
Okay cool! Go ahead and merge.
